cmake/sdp.cmake

```diff
@@ -49,3 +49,39 @@ function(sdp_assembly_generate hrt_srcs)
   add_dependencies(asm_gen syscall_list_h_target)
 
 endfunction()
+
+function(sdp_assembly_check hrt_srcs)
+  set(asm_check_msg "Checking if ASM files for Hard Real Time have changed.")
+
+  if(TARGET asm_check)
+    message(FATAL_ERROR "sdp_assembly_check() already called, please note that
+      sdp_assembly_check() must be called only once with a list of all source files."
+    )
+  endif()
+
+  add_custom_target(asm_check
+    COMMAND ${CMAKE_COMMAND} -D hrt_srcs=${hrt_srcs} -P ${ZEPHYR_NRF_MODULE_DIR}/cmake/sdp_asm_check.cmake
+    COMMAND_EXPAND_LISTS
```

Comment:
this does not work with lists.

First thing is that `;` will be replaced with `<space>`, thus resulting in: `-D hrt_srcs=foo.c bar.c`, and not `-D hrt_srcs="foo.c;bar.c"`.

To fix this, then first `COMMAND_EXPAND_LISTS` must be removed.
Secondly, `;` must be replaced by the `$<SEMICOLON>` generator expression.
and finally, the argument list must be quoted using `"`.